# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/riscv.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_riscv.rs.vanilla    (nonexistent)
+++ firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_riscv.rs    2020-10-16 11:37:13.522007639 +0200
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_riscv64.rs.vanilla	(nonexistent)
+++ firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_riscv64.rs	2020-10-16 11:37:13.522007639 +0200
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-94.0.1/third_party/rust/authenticator/build.rs.vanilla	2021-11-05 13:59:30.986672974 +0100
+++ firefox-94.0.1/third_party/rust/authenticator/build.rs	2021-11-05 14:01:27.454678982 +0100
@@ -35,6 +35,10 @@
         "ioctl_powerpc64le.rs"
     } else if cfg!(all(target_arch = "powerpc64", target_endian = "big")) {
         "ioctl_powerpc64be.rs"
+    } else if cfg!(target_arch = "riscv") {
+        "ioctl_riscv.rs"
+    } else if cfg!(target_arch = "riscv64") {
+        "ioctl_riscv64.rs"
     } else if cfg!(all(target_arch = "arm", target_endian = "little")) {
         "ioctl_armle.rs"
     } else if cfg!(all(target_arch = "arm", target_endian = "big")) {
--- firefox-94.0.1/third_party/rust/authenticator/src/linux/hidwrapper.rs.vanilla	2021-11-05 13:59:30.983672974 +0100
+++ firefox-94.0.1/third_party/rust/authenticator/src/linux/hidwrapper.rs	2021-11-05 14:00:32.606676153 +0100
@@ -32,6 +32,12 @@
 #[cfg(all(target_arch = "powerpc64", target_endian = "big"))]
 include!("ioctl_powerpc64be.rs");
 
+#[cfg(target_arch = "riscv")]
+include!("ioctl_riscv.rs");
+
+#[cfg(target_arch = "riscv64")]
+include!("ioctl_riscv64.rs");
+
 #[cfg(all(target_arch = "arm", target_endian = "little"))]
 include!("ioctl_armle.rs");
 
